examples/correlation_matrices/sampler.cpp

```diff
@@ -89,6 +89,26 @@ void write_to_file(std::string filename, std::vector<PointType> const& randPoint
     std::cout.rdbuf(coutbuf);
 }
 
+bool is_correlation_matrix(const MT& matrix){
+
+	const double tol = 1e-8;
+
+	//check if all the diagonal elements are ones
+	for(int i=0 ; i<matrix.rows() ; i++){
+		if(std::abs(matrix(i, i)-1.0) > tol){
+			return false;
+		}
+	}
+
+	//check if the matrix is positive definite
```

Comment:
Why not using the function  `isPositiveSemidefinite`. Note: that function is not using tolerance.

Reply:
I have updated the function and made use of the isPositiveSemidefinite function